pkg/metrics/builder.go

```diff
@@ -41,6 +43,12 @@ func (b *metricsConfigBuilder) WithPath(path string) *metricsConfigBuilder {
 	return b
 }
 
+//WithName specifies the name of the service
+func (b *metricsConfigBuilder) WithName(name string) *metricsConfigBuilder {
```

Comment:
```suggestion
func (b *metricsConfigBuilder) WithServiceName(name string) *metricsConfigBuilder {
```

Maybe this method name instead?

Reply:
After discussion, let's leave it as is for now and we can revisit later if it becomes confusing as to what is being named.